src/tools/perf/ucp_tests.cc

```diff
@@ -44,12 +44,13 @@ class ucp_perf_test_runner {
         iov_length_it = 0;
         for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
             iov[iov_it].buffer = (char *)buffer + iov_length_it;
-            iov[iov_it].length = m_perf.params.msg_size_list[iov_it];
+            iov[iov_it].dt     = ucp_dt_make_contig(m_perf.params.msg_size_list[iov_it]);
```

Comment:
let's move the implementation of iov datatype to next PR, (since it should also include a unit test), in this PR we can just rename 'length' to 'count' everywhere.
Another option is to update unit tests to use non-single-byte type in IOV tests